pkg/services/publicdashboards/service/service.go

```diff
@@ -238,6 +246,90 @@ func (pd *PublicDashboardServiceImpl) GetMetricRequest(ctx context.Context, dash
 	return metricReqDTO, nil
 }
 
+func (pd *PublicDashboardServiceImpl) GetAnnotations(ctx context.Context, reqDTO AnnotationsQueryDTO, accessToken string) ([]AnnotationEvent, error) {
+	_, dash, err := pd.GetPublicDashboard(ctx, accessToken)
+	if err != nil {
+		return nil, err
+	}
+
+	type annotationsDto struct {
+		Annotations struct {
+			List []DashAnnotation `json:"list"`
+		}
+	}
+
+	dto := annotationsDto{}
+	bytes, err := dash.Data.MarshalJSON()
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(bytes, &dto)
+	if err != nil {
+		return nil, err
+	}
+
+	anonymousUser, err := pd.BuildAnonymousUser(ctx, dash)
+	if err != nil {
+		return nil, err
+	}
+
+	uniqueEvents := make(map[int64]AnnotationEvent, 0)
+	for _, anno := range dto.Annotations.List {
+		// only consider enabled, grafana annotations
+		if anno.Enable && (*anno.Datasource.Uid == grafanads.DatasourceUID || *anno.Datasource.Uid == grafanads.DatasourceName) {
+			annoQuery := &annotations.ItemQuery{
+				From:         reqDTO.From,
+				To:           reqDTO.To,
+				OrgId:        dash.OrgId,
+				DashboardId:  dash.Id,
+				DashboardUid: dash.Uid,
+				Limit:        anno.Target.Limit,
+				MatchAny:     anno.Target.MatchAny,
+				SignedInUser: anonymousUser,
+			}
+
+			if anno.Target.Type == "tags" {
+				annoQuery.Tags = anno.Target.Tags
+			}
+
+			annotationItems, err := pd.AnnotationsService.Find(ctx, annoQuery)
+			if err != nil {
+				return nil, err
+			}
+
+			for _, item := range annotationItems {
+				event := AnnotationEvent{
+					Id:          item.Id,
+					DashboardId: item.DashboardId,
+					Tags:        item.Tags,
+					IsRegion:    item.TimeEnd > 0 && item.Time != item.TimeEnd,
+					Text:        item.Text,
+					Color:       *anno.IconColor,
+					Time:        item.Time,
+					TimeEnd:     item.TimeEnd,
+					Source:      anno,
+				}
+				if anno.Type == "dashboard" {
+					event.PanelId = item.PanelId
+				}
```

Comment:
If its a dashboard query, we include the panelId on the result so its only shown on that specific panel. If there's no panelId it will get shown on all the panels, which should only happen for tag and org annotations

Reply:
I should add a comment to indicate this, since I don't think it's clear